README.md

```diff
@@ -0,0 +1,116 @@
+# How to store trees with Django & PostgreSQL
+
+
+## Rationale
+
+If you ever had the need to store hierarchical data (trees) with Django, you
+probably had to use a library like
+[django-mptt](https://github.com/django-mptt/django-mptt) or
+[django-treebeard](https://github.com/django-treebeard/django-treebeard).
+Those libraries work fine at a small scale, but here at PeopleDoc we have
+encountered a lot of issues when using them at a bigger scale (tables with
+hundreds of thousands of rows and quite a lot of writings).
+
+It turns out that storing trees in a database a solved problem since a long
+time, at least with PostgreSQL. The
+[ltree](https://www.postgresql.org/docs/9.6/static/ltree.html) extension
+provides a convenient data structure, which is very fast on reads, and quite
+fast on writing. The algorithm used is very close to django-treebeard's
+materialized paths, but with all the power of PostgreSQL.
+
+The main downside of using ltree is that you have to maintain the materialized
+path yourself. It doesn't come with any tool to do it automatically.
+But fortunately, it's actually quite simple to maintain this path using
+PostgreSQL triggers!
+
+
+## Integration with Django
+
+In [`demo/categories/ltree.py`](/demo/categories/ltree.py) you will find a very
+simple Django field for the ltree data type. This field can be used in any
+Django model, and adds two lookups: `descendant` and `ancestor`. Those lookups
+allow you to query the descendants or the ancestors of any object with a very
+simple SQL query.
+
+For example, let's say you have the following model:
+```python
+from django.db import models
+
+from project.ltree import LtreeField
+
+
+class Category(models.Model):
+  parent = models.ForeignKey('self', null=True)
+  code = models.CharField(maxlength=32, unique=True)
```

Comment:
wow, is this limiting the size of the tree on the amount of levels you can fit in a 32 chars code ? :D A rather bold choice :D

Reply:
Nope, here code is the unique identifier of each node. There is no limit on the total as `path` is a text field.